src/common/serviceKeys.ts

```diff
@@ -0,0 +1,16 @@
+import { ServiceKeys } from './types'
+import { pickRandom } from './utils'
+
+export function getRandomServiceKey(
+  serviceKeys: ServiceKeys,
+  index: string
+): string | undefined {
+  const keys = serviceKeys[index]
+  if (keys == null || keys.length === 0) return undefined
+  return pickRandom(keys, 1)[0]
+}
+
+export function getServiceKeyIndex(url: string): string {
+  const host = new URL(url, `https://${url}`).host
+  return host
```

Comment:
While you're at the rebase. Feel free to change this to:

```ts
const url =  = new URL(url, `https://${url}`)
return url.host
```

Just so that way we can easily insert a `console.log(url)` at a later point for debugging.